api/functions/graphql/types/post.js

```diff
@@ -37,7 +40,17 @@ const Author = objectType({
     definition(t) {
         t.nonNull.int('id');
         t.nonNull.string('name');
-        t.nonNull.string('avatar');
+        t.nonNull.string('avatar', {
+            async resolve(parent) {
+                const imgObject = await prisma.hostedImage.findUnique({
+                    where: {
+                        id: parent.avatar_id
+                    }
+                });
+
```

Comment:
This method of querying the hosted image works well.
However, in the case of nested resolvers, it's generally better to use prisma's fluentAPI to fetch the relations fields.
The reason for that is: prisma batches these queries internally, so they becomes faster than if they were done as normal queries.

To read more about this idea, you can check this page in their docs:
https://www.prisma.io/docs/guides/performance-and-optimization/query-optimization-performance#solving-the-n1-problem

I've made this change to the `avatar` field of the User, and the `cover_image` field of the story, to make it clearer about what I meant.